src/Sema.zig

```diff
@@ -17157,6 +17157,36 @@ fn zirAsm(
             .ComptimeInt => arg.* = try sema.coerce(block, Type.usize, uncasted_arg, src),
             .ComptimeFloat => arg.* = try sema.coerce(block, Type.f64, uncasted_arg, src),
             else => {
+                const input_op_src = block.src(.{
+                    .asm_input_op = .{
+                        .asm_node_offset = extra.data.src_node,
+                        .input_index = @intCast(arg_i),
+                    },
+                });
+                if (try sema.typeRequiresComptime(uncasted_arg_ty)) {
+                    return sema.fail(
+                        block,
+                        input_op_src,
+                        "type '{}' is comptime-only and cannot be used for an assembly input operand",
+                        .{uncasted_arg_ty.fmt(pt)},
+                    );
+                }
```

Comment:
I seem to recall something about the SPIR-V backend using types as asm inputs... yeah, here we are, #20550 covers it. @Snektron, I'm a little unclear from that proposal -- does this need to work today to avoid regressing stuff? If so, we'll need to add a special case here for SPIR-V.

Reply:
It's still a proposal so I assume it's not implemented yet so this can be changed when that proposal gets implemented? Well let's wait for an answer from @Snektron.

Reply:
So would I do this?
```zig
                if (try sema.typeRequiresComptime(uncasted_arg_ty)) {
                    const target = pt.zcu.getTarget();
                    const arch = target.cpu.arch;
                    const is_spirv = arch.isSpirV();
                    if (uncasted_arg_ty.zigTypeTag(zcu) != .Type and !is_spirv) {
                        return sema.fail(
                            block,
                            input_op_src,
                            "type '{}' is comptime-only and cannot be used for an assembly input operand",
                            .{uncasted_arg_ty.fmt(pt)},
                        );
                    }
                }
```

Reply:
`or`, not `and`! Aside from that, looks good.

Reply:
Its not properly implemented in SPIR-V yet in any case, feel free to do whatever for now. We'll make the necessary changes when its required.

Sorry for the late reply :sweat_smile: